tests/Signature/SignatureV4Test.php

```diff
@@ -220,6 +220,83 @@ public function testEnsuresContentSha256CanBeCalculatedWhenSeekFails()
         $signature->signRequest($request, $credentials);
     }
 
+    public function testUnsignedPayloadProvider()
```

Comment:
these are all test cases with `https`, maybe also add test for `http` and verify it follows regular V4